### ISSUE TITLE
Error parsing query in a transaction

### ISSUE THREAD

**Issue by alpeshgaglani** [orientechnologies/orientdb#5397]
Hi,
I am getting an error for a query that used to work in 2.0.x, but is now giving a parsing error (with 2.1.6 on Ubuntu 14.04).
The db looks something like:
```
create class A extends V
create class AV extends V
create class Has_AV extends E
create property Has_AV.isPublished Boolean
```
A can have many outgoing edges to AV, and only one of them could have isPublished = true. 
Our query that fails is:
```
script sql
begin
let $av = select from #20:0
let publishedVersionEdge = select $pvEdge[0] as row from $av.in_HasAV.out let $pvEdge = (selec
t from $current.out_HasAV where isPublished = true)
let unpublishedEdge = update $publishedVersionEdge.row set isPublished = false
end
```
For this query, we get the following error:
```
Error: com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #0: Error
 parsing query:
update $publishedVersionEdge.row set isPublished = false
Encountered " "." ". "" at line 1, column 29.
Was expecting one of:
    <SET> ...
    <ADD> ...
    <PUT> ...
    <MERGE> ...
    <CONTENT> ...
    <REMOVE> ...
    <INCREMENT> ...
```

I've tried any number of combinations to make this work over the past couple days, and am finally giving in :). Please help!

Thanks,
Alpesh

**Comment by luigidellaquila** [comment]
Hi @alpeshgaglani 

I think it's not supported by the new parser (sometimes corner cases like this still arise), I'll fix it soon.
In the meantime, could you please try to work around it with a subquery?
eg.
```
let unpublishedEdge = update (select expand(row) from $publishedVersionEdge) set isPublished = false
```
This should work fine

Thanks

Luigi

**Comment by alpeshgaglani** [comment]
I had ended up with a similar solution, and it works fine. Is there some documentation on what's supported in the new parser? I ask because I want us to go through all of our existing queries and fix things that might not be supported before initiating an exhaustive test run. 

Thanks SO much for the quick response!

**Comment by luigidellaquila** [comment]
Hi @alpeshgaglani 

the new parser is supposed to support all the syntax of the old one, so you can refer to doc pages for single SQL statements. 

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @alpeshgaglani 

I fixed this problem in both 2.1.x and develop branches.
I'm closing this issue, please reopen in case you experience any probelms

Thanks

Luigi
